plugins/pulp_docker/plugins/importers/importer.py

```diff
@@ -389,9 +419,66 @@ def _remove_manifest(cls, repo, manifest):
         :param manifest: The Manifest being removed
         :type  manifest: pulp_docker.plugins.models.Manifest
         """
+        #TODO Before removing the image manifest, add the check
+        # if this manifest is referenced in manifest list
         cls._purge_unlinked_tags(repo, manifest)
         cls._purge_unlinked_blobs(repo, manifest)
 
+    @classmethod
+    def _remove_manifest_list(cls, repo, manifest):
+        cls._purge_unlinked_manifests(repo, manifest)
+        cls._purge_unlinked_tags(repo, manifest)
+
+    @staticmethod
+    def _purge_unlinked_manifests(repo, manifest):
```

Comment:
I don't feel good about calling a manifest list simply a "manifest", like it's just another kind of manifest. I know docker is blurring the line a bit, but do you think it's reasonable for us to be very clear about differentiating a manifest list from manifests?